apache/trafficcontrol
# Support CentOS 8 RPMs

## What does this PR (Pull Request) do?


- [x] This PR fixes #5319
- Adds the option to build for either CentOS 7 or CentOS 8. CentOS 7 is the default, except for CDN in a Box, which uses CentOS 8 RPMs by default.
- Upgrades the builder Docker images to CentOS 8, although they can still be built for CentOS 7.

This PR builds off of @hbeatty's CentOS 8 progress from #5137.

## Which Traffic Control components are affected by this PR?


- CDN in a Box
- Documentation
- Build system

## What is the best way to verify this PR?

1. Check that Traffic Control works with RPMs built on and targeting CentOS 8
    ```shell script
   make build-builders -j6
   docker-compose build --parallel
   docker-compose up
    ```

2. Check that Traffic Control works with RPMs built on and targeting CentOS 7 
    ```shell script
   export RHEL_VERSION=7
   make build-builders -j6
   docker-compose build --parallel
   docker-compose up
    ```

## The following criteria are ALL met by this PR


- [x] Existing tests are sufficient, no additional tests are unnecessary

## Additional Information

Note that only Go-based projects contain binaries that `elfdeps` checks the linking for. The Go binaries have been statically linked since #4758, so that check does not find any dynamic section in the binaries, and we can use CentOS 8 builders to build RPMs targeting either CentOS 7 (the default for now) or CentOS 8.
